enhancements/cluster-logging/cluster-logging-es-pipeline-processing.md

```diff
@@ -0,0 +1,197 @@
+
+Over the time we have gathered number of requests from customers asking for possibility to process additional data. See <https://issues.redhat.com/browse/LOG-785>.
+
+For the purpose of this proposal we have already investigated various approaches (and their pros and cons) of storing additional JSON data in Elasticsearch indices as part of <https://issues.redhat.com/browse/LOG-842>.
+
+
+### Goals
+
+The high-level goal of this proposal is to lay the groundwork to enable approaches B) and C) as discussed in [LOG-842](https://issues.redhat.com/browse/LOG-842).
+
+Quick comparison of approaches B) and C):
+
+- Approach B) Whenever user needs to store data with a new data model we solve it by pointing this data to new index (or to existing index if there is already one with exactly the same data model). This is very comfy approach for the user except it can lead to very high number of indices and querying experience can be complicated because there are very little or no common conventions of document fields meaning and types.
+
+- Approach C) is on the other side of the spectrum. All data is stored into common index and whenever user needs to store data with a new data model then it is necessary to transfor the data to fit existing data model in existing index. This approach requires attention to data model and users may need to invest time and resources to transform new data sources. On the other hand there is minimum additional indices created in ES cluster and data can be queried uniformly because document fields have predictable meaning and types.
+
+It should be pointed out that Elasticsearch pipeline processes can support both cases. Depending on how much users are willing to pay attention to their data model they can start with approach B) and ventually transition to approach C) once they feel more confident about managing their own data model across more log resources.
+
+In this proposal we assume that the approach B is the one we would start enabling first. But nothing is stopping us from enabling the approach C in the future. Though it is fair to say that approach C will be more complicated to implement because it will require management of custom processing pipelines.
+
+We will be successful when:
+
+* An OCP user (or admin only?)  is able to configure specific logs to be pointed into different (non-default) index in Elasticsearch based on the data contained in the document itself.
+* An OCP user (or admin only?) can provide custom index mapping for the data. This is specifically needed if the [default Elasticsearch type inference mechanism](https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-field-mapping.html) does not lead to acceptable results.
```

Comment:
I would probably see this as a nice to have when we go a define the scope for a first deliverable. This definitely makes the experience better but it does not break if its not there as your JSON logs still get indexed. WDYT?

Reply:
Agree, I just want to make sure we keep this in mind when designing things so that we do not close the door for us if we decide to implement this in the future. I will consider to move this point to different category ("optional" or "also consider" kind of category).

Reply:
Point moved to optional category.